src/stream/src/executor/source/reader_stream.rs

```diff
@@ -214,10 +215,22 @@ impl StreamReaderBuilder {
                             actor_id = self.actor_ctx.id,
                             "stream source reader error"
                         );
+                        is_error = true;
                         break 'consume;
                     }
                 }
             }
+            if !is_error {
```

Comment:
Is there any other cases where the connector read stream can be finite (that is, finishes without error)?